### PR TITLE
Document generateCompileCommands task better

### PR DESCRIPTION
**Pull request by spacey-sooty**
includes a warning about not being reccomended for most users

### CODE REVIEW

**Comment by spacey-sooty** [comment]
\inspector fix all

File: source/docs/software/advanced-gradlerio/gradlew-tasks.rst [hunk 1]
```diff
@@ -13,7 +13,13 @@ Build tasks
 CompileCommands tasks
 ---------------------
 
-``./gradlew generateCompileCommands`` - Generate compile_commands.json for C++ programs. This is a configuration file that is supported by many Integrated Development Environments and build tools.
+``./gradlew generateCompileCommands`` - Generate compile_commands.json for C++ programs. This is a configuration file that is used by clangd, which is supported by many Integrated Development Environments and build tools.
+
+.. warning::
+    Using compile commands isn't recommended for most users and has a variety of pitfalls compared to in built intellisense.
```
Comment (rzblue) [inline source/docs/software/advanced-gradlerio/gradlew-tasks.rst:19]:
```suggestion
    Using compile commands isn't recommended for most users and has a variety of pitfalls compared to the built-in Intellisense.
```

File: source/docs/software/advanced-gradlerio/gradlew-tasks.rst [hunk 2]
```diff
@@ -13,7 +13,13 @@ Build tasks
...
+
+.. note::
+    This file will be outputted under ``build/TargetedCompileCommands/targetplatformandbuildtype`` but needs to be moved to the root of your project. Make sure you have gitignored this file.
```
Comment (rzblue) [inline source/docs/software/advanced-gradlerio/gradlew-tasks.rst:22]:
```suggestion
    This file will be saved in ``build/TargetedCompileCommands/targetplatformandbuildtype`` but needs to be moved to the root of your project. Make sure you have gitignored this file.
```
Also, most IDEs have an option to specify compile command file location

**Review by rzblue — CHANGES_REQUESTED** [review]

File: source/docs/software/advanced-gradlerio/gradlew-tasks.rst [hunk 3]
```diff
@@ -13,7 +13,13 @@ Build tasks
 CompileCommands tasks
 ---------------------
 
-``./gradlew generateCompileCommands`` - Generate compile_commands.json for C++ programs. This is a configuration file that is supported by many Integrated Development Environments and build tools.
+``./gradlew generateCompileCommands`` - Generate compile_commands.json for C++ programs. This is a configuration file that is used by clangd, which is supported by many Integrated Development Environments and build tools.
+
+.. warning::
+    Using compile commands isn't recommended for most users and has a variety of pitfalls compared to the built-in intellisense.
+
+.. note::
```
Comment (sciencewhiz) [inline source/docs/software/advanced-gradlerio/gradlew-tasks.rst:21]: I think this is better off as part of the description rather a note

**Review by sciencewhiz — APPROVED** [review]

**Review by rzblue — APPROVED** [review]

**Pull request merged**